subsys/net/lib/download_client/src/download_client.c

```diff
@@ -422,6 +422,8 @@ void download_thread(void *client, void *a, void *b)
 				continue;
 			}
 			if (rc < 0) {
+				/* Something was wrong with the header */
+				error_evt_send(dl, EINVAL);
```

Comment:
I'm open to suggestions of better errno values.  I suppose the really smart thing to do would be to have header_parse() return an errno value specific to what it actually found in the header, converting 404 to ENONET, 401 to EACCESS, etc, but that is possibly overkill for this problem.